code/datums/elements/pet_bonus.dm

```diff
@@ -0,0 +1,35 @@
+/**
+ * # Pet bonus element!
+ *
+ * Bespoke element that plays a fun message, sends a heart out, and gives a stronger mood bonus when you pet this animal.
+ * I may have been able to make this work for carbons, but it would have been interjecting on some help mode interactions anyways.
+ */
+/datum/element/pet_bonus
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+
+	///optional cute message to send when you pet your pet!
+	var/emote_message
+
+/datum/element/pet_bonus/Attach(datum/target, emote_message)
+	. = ..()
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.emote_message = emote_message
+	RegisterSignal(target, COMSIG_ATOM_ATTACK_HAND , .proc/on_attack_hand)
+
+/datum/element/pet_bonus/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, COMSIG_ATOM_ATTACK_HAND)
+
+/datum/element/pet_bonus/proc/on_attack_hand(mob/living/pet, mob/living/petter)
+	SIGNAL_HANDLER
+
+	if(pet.stat != CONSCIOUS || petter.combat_mode)
+		return
+
+	new /obj/effect/temp_visual/heart(pet.loc)
+	if(emote_message && prob(33))
+		pet.manual_emote("[emote_message]")
```

Comment:
Am I missing something; why can't you just pass the `emote_message` argument directly to the function call, rather than taking the string representation of the string?

Reply:
Actually you probably could, lemmie do that.